Content.Server/Chat/SuicideSystem.cs

```diff
@@ -53,9 +57,16 @@ public bool Suicide(EntityUid victim)
             if (suicideEvent.AttemptBlocked)
                 return false;
 
+            if (!_gameTicker.OnGhostAttempt(mindId, false, mind: mind))
+                return false;
+
             DefaultSuicideHandler(victim, suicideEvent);
```

Comment:
```                        
                        var ev = new ActiveTimerTriggerEvent(grenade, uid);
                        RaiseLocalEvent(uid, ref ev);
```

do something like this rather than calling the event handler directly, move the `ApplyLethalDamage()` to the suicide handler so that it properly acts as a fallback when the Event is not handled by other systems